### PR TITLE
Add TRIVY_DB_REPOSITORY env to vulnerabilitiy scan

### PR DESCRIPTION
**Pull request by gigxz**
## _Merging this PR_
- use the squash-merge strategy

## Description
[//]: # (Summarize changes and include links related issue. List any new dependencies)

https://greenriver.slack.com/archives/C03GCTC2C/p1729271527659529?thread_ts=1729098847.019139&cid=C03GCTC2C
May address failures in vulnerability scanning

## Type of change
[//]: # (e.g., Bug fix, New feature, Code clean-up, Dependency update)

## Checklist before requesting review
[//]: # (Remove any items that are not applicable)
- [x] I have performed a self-review of my code
- [ ] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [ ] If adding a new endpoint / exposing data in a new way, I have:
  - [ ] ensured the API can't leak data from other data sources
  - [ ] ensured this does not introduce N+1s
  - [ ] ensured permissions and visibility checks are performed in the right places
- [ ] I have updated the documentation (or not applicable)
- [ ] I have added spec tests (or not applicable)
- [ ] I have provided testing instructions in this PR or the related issue (or not applicable)

### CODE REVIEW

File: .github/workflows/build_images.yml [hunk 1]
```diff
@@ -162,6 +162,8 @@ jobs:
       - name: Run vulnerability scan
         id: vuln_scan
         uses: aquasecurity/trivy-action@0.14.0
+        env:
+          TRIVY_DB_REPOSITORY: "public.ecr.aws/aquasecurity/trivy-db:2"
```
Comment (gigxz) [inline .github/workflows/build_images.yml:166]: the default is `ghcr.io/aquasecurity/trivy-db`. this gets it from here https://gallery.ecr.aws/aquasecurity/trivy-db instead which may be more reliable

Comment (eanders) [reply .github/workflows/build_images.yml:166]: Hmm, seems this failed as well.  @yakloinsteak is this a different error?

Comment (yakloinsteak) [reply .github/workflows/build_images.yml:166]: Try upgrading the trivy action version to 0.28.0. I'll take a look if that fails.

Comment (gigxz) [reply .github/workflows/build_images.yml:166]: Updated

Comment (eanders) [reply .github/workflows/build_images.yml:166]: @yakloinsteak still failing.

**Comment by yakloinsteak** [comment]
Add 
```
TRIVY_JAVA_DB_REPOSITORY: "public.ecr.aws/aquasecurity/trivy-java-db:1"
```
as well

**Comment by gigxz** [comment]
I _think_ it worked this time and is just failing on actual vulnerabilities https://github.com/greenriver/hmis-warehouse/actions/runs/11501786374/job/32016315796

**Comment by eanders** [comment]
Yes, I'm working through the vulnerability now.

**Comment by eanders** [comment]
@gigxz , @yakloinsteak fixed it (again.)  I think we're good to merge this, you ok with that?

**Review by eanders — APPROVED** [review]

**Pull request merged**